common/defs: increase maxfee_kb + update script

mmilata opened the pull request:
Updates `maxfee_kb` so that the maximum fee per transaction is approximately 10 USD, for bitcoin-based altcoins except bitcoin. Fixes #1192.

`maxfee.py` is provided to update the definitions based on current price data from coinmarketcap.com. It should be easy to extend it with a check mode that could be run from nightly CI and fail the pipeline if a fee exceeds some metric.

tsusanka commented:
The only thing I am a bit worried is that we assume that the "traditional" transaction has 250B. But we assume that based on our knowledge of Bitcoin but how do we know that the same applies for the altcoins? But that being said - as I went through the numbers it seems to me there is not a single instance where we _lower_ the fee. Is that correct? So then this should be just fine, or in another words not a regression.

matejcik reviewed (commented):
existing changes LGTM

but there should be something -- either in `maxfee.py` itself or in some wrapper script -- that does The Right Thing to refresh this.

iow, perhaps if no filenames are provided, pick everything in `coins/bitcoin` that is not `"Bitcoin", "Testnet", "Regtest"`, and does not contain "testnet" in its name?

matejcik commented:
Also I'm wondering if we should make this more stable? So that repeated runs do not introduce irrelevant changes.
Choose a reasonable rounding limit so that if the price of a given coin doesn't change by more than, say, 20 % ? then the maxfee does not change either.
and emit a warning if the maxfee jumped significantly (more than rounding*10 ?)

tsusanka commented:
I am not sure we need to worry about that. Running the script once a year seems just fine? I don't think we can expect the altcoins to increase their prices 10x times.

matejcik commented:
> I don't think we can expect the altcoins to increase their prices 10x times.

that's the point :)

in any case rounding would still be good; it would make reviewing the diffs easier.

mmilata commented:
> 250B. But we assume that based on our knowledge of Bitcoin but how do we know that the same applies for the altcoins? But that being said - as I went through the numbers it seems to me there is not a single instance where we lower the fee. Is that correct?

Yeah, it's a guess. We'd need to keep track of usual transaction size for each coin to get more precise results. Also correct that no fee decreased (the one for bitcoin did but that's not included).

@matejcik I've changed the script to update to everything that's not bitcoin/regtest/testnet, PTAL. Also implemented rounding that keeps only the first 2 leading digits non-zero, if I'm not mistaken the result should change when the number changes by roughly 10%?

Gonna go over this one more time, some of the new numbers are surprisingly high ...

mmilata commented:
Ok, I checked that the new `maxfee_kb` values are consistent with the table in #1192. The rounding algorithm and price-jump warnings can probably be improved but I'm afraid we can't expect much stability in the altcoin prices either way.

tsusanka reviewed (approved):
Please rebase and force-push!

mmilata commented:
Rebased.
